plone/outputfilters/filters/resolveuid_and_caption.py

```diff
@@ -157,6 +157,44 @@ def __call__(self, data):
                     actual_url = relative_root.absolute_url()
                     href = urljoin(actual_url + '/', subpath) + appendix
                 attributes['href'] = href
+        for elem in soup.find_all(['source', 'iframe']):
+            # SOURCE is used for video and audio.
+            # IFRAME is used to embed PDFs.
+            attributes = elem.attrs
+            attrn = 'src'
+            src = attributes.get(attrn)
+            # an 'a' anchor element has no href
+            if not src:
+                attrn = 'srcset'
+                src = attributes.get(attrn)
+                if not src:
+                    continue
+            url_parts = urlsplit(src)
```

Comment:
Does the parse work on an srcset with more than one url? I am pretty sure not. I think we need some additional tests in ``test_resolveuid_and_caption.py``

Reply:
I have added an additional test and handled all the documented `srcset=` cases now.